sssd 2.13.1 FTBFS oidc

"Regression" from 2.13.0 to 2.13.1:

```
./configure --host=x86_64-suse-linux --build=x86_64-suse-linux --program-prefix= --disable-dependency-tracking --prefix=/usr 
--exec-prefix=/usr --bindir=/usr/bin --sbindir=/usr/sbin --sysconfdir=/etc --datadir=/usr/share --includedir=/usr/include --libdir=/usr/
lib64 --libexecdir=/usr/libexec --localstatedir=/var --sharedstatedir=/var/lib --mandir=/usr/share/man --infodir=/usr/share/info --with-
db-path=/var/lib/sss/db --with-pipe-path=/var/lib/sss/pipes --with-pubconf-path=/var/lib/sss/pubconf --with-gpo-cache-path=/var/lib/sss/
gpo_cache --with-environment-file=/etc/sysconfig/sssd --with-initscript=systemd --with-syslog=journald --with-pid-path=/run/sssd --enabl
e-pammoddir=/usr/lib64/security --with-ldb-lib-dir=/usr/lib64/samba/ldb --disable-ldb-version-check --without-python2-bindings --without
-oidc-child --with-sssd-user=sssd --with-vendordir=/usr/etc/sssd --with-selinux=yes --with-subid --with-os=suse

## --without-oidc-child (!)

[   79s] gcc -DHAVE_CONFIG_H -I.  -Wall -I.. -I./src/sss_client -I./src -I.     -I/usr/include/samba-4.0 -I/usr/include/dbus-1.0 -I/usr>
[   79s] src/oidc_child/libcrypto/oidc_child_get_jwk.c:32:10: fatal error: jose/b64.h: No such file or directory
[   79s]    32 | #include "jose/b64.h"
[   79s]       |          ^~~~~~~~~~~~
[   79s] compilation terminated.
```

--- a/Makefile.am
+++ b/Makefile.am
@@ -3634,14 +3634,12 @@ test_cert_utils_SOURCES = \
     src/tests/cmocka/test_cert_utils.c \
     src/responder/ssh/ssh_cert_to_ssh_key.c \
     src/util/cert_derb64_to_ldap_filter.c \
-    src/oidc_child/libcrypto/oidc_child_get_jwk.c \
     $(NULL)
 test_cert_utils_CFLAGS = \
     $(AM_CFLAGS) \
     -U SSSD_LIBEXEC_PATH -DSSSD_LIBEXEC_PATH=\"$(abs_builddir)\" \
     -I$(abs_builddir)/src \
     $(CRYPTO_CFLAGS) \
-    $(JOSE_CFLAGS) \
     $(CMOCKA_CFLAGS) \
     $(NULL)
 test_cert_utils_LDADD = \
@@ -3649,13 +3647,17 @@ test_cert_utils_LDADD = \
     $(POPT_LIBS) \
     $(TALLOC_LIBS) \
     $(CRYPTO_LIBS) \
-    $(JOSE_LIBS) \
     libsss_debug.la \
     libsss_test_common.la \
     libsss_cert.la \
     libsss_crypt.la \
     libsss_certmap.la \
     $(NULL)
+if BUILD_OIDC_CHILD
+    test_cert_utils_SOURCES += src/oidc_child/libcrypto/oidc_child_get_jwk.c
+    test_cert_utils_CFLAGS += $(JOSE_CFLAGS)
+    test_cert_utils_LDADD += $(JOSE_LIBS)
+endif
 
 test_data_provider_be_SOURCES = \
     src/providers/data_provider_be.c \

--- a/src/tests/cmocka/test_cert_utils.c
+++ b/src/tests/cmocka/test_cert_utils.c
@@ -32,7 +32,9 @@
 #include "tests/cmocka/common_mock.h"
 #include "util/crypto/sss_crypto.h"
 #include "responder/ssh/ssh_private.h"
+#ifdef BUILD_OIDC_CHILD
 #include "oidc_child/oidc_child_util.h"
+#endif
 
 #ifdef HAVE_TEST_CA
 #include "tests/test_CA/SSSD_test_cert_pubsshkey_0001.h"
@@ -830,6 +832,7 @@ void test_cert_to_ssh_2keys_with_certmap_2_send(void **state)
     talloc_free(ev);
 }
 
+#ifdef BUILD_OIDC_CHILD
 void test_get_jwk_from_pkcs12(void **state)
 {
     int ret;
@@ -871,6 +874,7 @@ void test_get_jwk_from_pkcs12(void **state)
     talloc_free(pkcs12);
     talloc_free(jwk);
 }
+#endif
 
 int main(int argc, const char *argv[])
 {
@@ -911,8 +915,10 @@ int main(int argc, const char *argv[])
                                         setup, teardown),
         cmocka_unit_test_setup_teardown(test_cert_to_ssh_2keys_with_certmap_2_send,
                                         setup, teardown),
+#ifdef BUILD_OIDC_CHILD
         cmocka_unit_test_setup_teardown(test_get_jwk_from_pkcs12,
                                         setup, teardown),
+#endif
 #endif
     };
 

